app/src/main/java/info/nightscout/android/medtronic/service/MedtronicCnlIntentService.java

```diff
@@ -232,12 +270,36 @@ protected void onHandleIntent(Intent intent) {
                     cnlReader.updatePumpStatus(pumpRecord);
 
                     cnlReader.endEHSMSession();
+                    MainActivity.dbgCNL_beginEHSMSession -= 1;
 
                     if (pumpRecord.getSgv() != 0) {
+
+                        String sgvString;
+                        if (MainActivity.mmolxl) {
+                            sgvString = MainActivity.sgvFormatter.format((float) pumpRecord.getSgv() / MainActivity.MMOLXLFACTOR);
```

Comment:
This class should always return mg/dL (yes, even though I'm in Australia and we use mmol/L, too). That's because this is the units that the device uses. Any conversion to other units for display purposes should be done in the MainActivity itself. This also means that sgvFormatter isn't public static.